jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ResourceSelector.kt

```diff
@@ -74,7 +61,7 @@ class ResourceSelector<T> private constructor(
 
     @JvmOverloads
     @Synchronized
-    fun reload(forceFetch: Boolean = false) {
+    fun reload(forceFetch: Boolean = false, awsConnectionOverride: () -> Pair<AwsRegion, ToolkitCredentialsProvider> = awsConnection) {
```

Comment:
Why can't this be hidden inside the `awsConnection` implementation?

Reply:
I'm not sure I understand what you mean - this is so that reload can be invoked on an existing selector to re-fetch using new connection settings.

So the goal here is to allow an optional parameter (or default to the one the REsourceSelector was initialized with

Reply:
My point is why have both - the 'default' parameter is a function call - so the instantiator can control what gets returned there at any time.

Reply:
Discussed offline, on the same page